accounts/abi/event_test.go

```diff
@@ -155,6 +155,16 @@ func TestEventTupleUnpack(t *testing.T) {
 		jsonEventPledge,
 		"",
 		"Can unpack Pledge event into slice",
+	}, {
+		pledgeData1,
+		&[3]interface{}{&common.Address{}, &bigint, &[3]byte{}},
+		&[3]interface{}{
+			&addr,
+			&bigintExpected2,
+			&[3]byte{'u', 's', 'd'}},
+		jsonEventPledge,
+		"",
+		"Can unpack Pledge event into an array",
```

Comment:
I think I wasn't clear. This is a good test to have but it isn't exactly what I meant when I said add a test for a static array. Relook at what you have written and consider the following problem.

Let's assume we create an event of the signature `event A(uint[3] indexed a, uint[3] b, string c)`. Currently it is my belief that your current implementation will fail this test and overflow.